pkg/noderesourcetopology/match.go

```diff
@@ -0,0 +1,358 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/klog/v2"
+	v1qos "k8s.io/kubernetes/pkg/apis/core/v1/helper/qos"
+	bm "k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/bitmask"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	apiconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	topoclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	topoinformerexternal "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	topologyinformers "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	topoinformerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions/topology/v1alpha1"
+)
+
+const (
+	// NodeResourceTopologyMatchName is the name of the plugin used in the plugin registry and configurations.
+	NodeResourceTopologyMatchName = "NodeResourceTopologyMatch"
+)
+
+var _ framework.FilterPlugin = &NodeResourceTopologyMatch{}
+
+type nodeTopologyMap map[string]topologyv1alpha1.NodeResourceTopology
+
+type PolicyHandler interface {
```

Comment:
Now we have 2 implementation, they are not similar, one is working per all resources of all containers and makes decision for whole pod, another one makes decision per container. And quite different implementation may exist in future, even not related to NUMA, since we may also support https://github.com/intel/cri-resource-manager (by using the same noderesourcetopology-api, it's now flexible enough to represent any kind of hardwares)

Since topologyPolicy is an attribute of the worker node, the idea behind this is to choose appropriate handler for node's policy or skip node if policy is not found (e.g. wasn't provided) for node.
It could be done simpler e.g. with _if_ clause in case of one Filter plugin per all policies we will have. Or one Filter plugin per each policy, but in this case each implementation of Filter plugin will do unnecessary job like skipping nodes if node configured for another policy.

Reply:
Commented in https://github.com/kubernetes-sigs/scheduler-plugins/pull/143#discussion_r570570966